src/permissions/permissions.service.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import { Injectable } from '@nestjs/common';
+import { User } from '../users/user.entity';
+import { Note } from '../notes/note.entity';
+import { ConsoleLoggerService } from '../logger/console-logger.service';
+
+@Injectable()
+export class PermissionsService {
+  constructor(private readonly logger: ConsoleLoggerService) {
+  }
+  mayRead(user : User, note : Note) : boolean {
+    if (this.isOwner(user, note)) return true;
+
+    if (this.hasPermissionUser(user, note, false)) return true;
+
+    if (this.hasPermissionUser(user, note, false)) return true;
```

Comment:
Despite this PR being still a draft, I want to point out this copy-and-paste error in case you didn't catch it.

```suggestion
    if (this.hasPermissionGroup(user, note, false)) return true;
```

Reply:
fixed